contrib/brad/pybrad.cxx

```diff
@@ -4,22 +4,106 @@
 #include <pybind11/stl.h>
 #include <pybind11/numpy.h>
 
+#include <stdexcept>
+#include <string>
 #include <memory>
 #include <sstream>
 #include <vector>
 #include <array>
 
+
+#include <vil/vil_convert.h>
 #include <vil/vil_image_view.h>
+#include <vnl/vnl_math.h>
 #include <brad/brad_image_atmospherics_est.h>
 #include <brad/brad_image_metadata.h>
+#include <brad/brad_calibration.h>
 
 namespace py = pybind11;
 
 namespace pyvxl { namespace brad {
 
+vil_image_view<float> estimate_reflectance(vil_image_view<float> const& radiance_img,
+                                           brad_image_metadata const& mdata,
+                                           float mean_reflectance,
+                                           bool average_airlight,
+                                           bool is_normalize)
+{
+
+  if (radiance_img.pixel_format() != VIL_PIXEL_FORMAT_FLOAT) {
+    throw std::invalid_argument("ERROR: vxl.contrib.brad.estimate_reflectance: expecting floating point radiance image\n");
+  }
+
+  unsigned int ni = radiance_img.ni();
+  unsigned int nj = radiance_img.nj();
+  unsigned int np = radiance_img.nplanes();
+  if (mean_reflectance <= 0.0)
+    is_normalize = false;
+
+  auto* reflectance_img = new vil_image_view<float>(ni, nj, np);
+  bool success = brad_estimate_reflectance_image(radiance_img, mdata, mean_reflectance, *reflectance_img, average_airlight, is_normalize);
+
+  if (!success)
+    throw std::runtime_error(std::string("ERROR: vxl.contrib.brad.estimate_reflectance: brad_estimate_reflectance_image failed.\n"));
+
+  return *reflectance_img;
+}
+
+/* std::tuple<unsigned int, > get_image_footprint() */
```

Comment:
If junk code, remove
Also C style comments?! What kind of editor are you USING!?

Reply:
Removed. I'm using the vim-commentary plugin :D